fox_scraper/fox_scraper.py

```diff
@@ -46,41 +46,41 @@ def runner(self):
         :return: a list of items (Item class) of the entire catalog
         """
-        main_categories_links = self.get_main_categories()
+        self.get_main_categories()
         catalog = []
-        for main_cat_num in range(len(main_categories_links)):
-            sub_categories = self.get_sub_categories(main_categories_links[main_cat_num], main_cat_num)
-            category_catalog = self.get_category_catalog(sub_categories, main_cat_num)
+        for main_category in self.main_categories_dict:
+            self.get_sub_categories(main_category)
+            category_catalog = self.get_category_catalog(main_category)
             catalog.append(category_catalog)
-        return catalog
+        self.send_all_items(catalog)
 
     def get_main_categories(self):
         """
         this function will scrap all main catalog categories
+        this function will scrap all main catalog categories
         :return: a list of beutifulSoup objects
         """
         main_categories_content = self.get_parsed_web_content(self.site_address)
         main_categories_links = []
         for link in main_categories_content.find_all(class_='link padding_hf_v '):
             main_categories_links.append(link.get('href'))
-            self.main_categories.append(link.get_text().strip())
-        return main_categories_links
+            self.main_categories_dict.update({str(link.get_text().strip()): link.get('href')})
 
-    def get_sub_categories(self, category_link, category_num):
+    def get_sub_categories(self, main_category):
         """
         this function will scrap sub-catalog links per category
-        :param category_link: sub-category link from the site
-        :param category_num: category number for item description
+        :param main_category: category link from the site
         :return: a list of sub-categories link per main-category
         """
-        sub_categories_content = self.get_parsed_web_content(str(category_link))
+        category_link = self.main_categories_dict[main_category]
+        sub_categories_content = self.get_parsed_web_content(category_link)
+
         sub_categories = []
-        self.sub_categories.append([])
         for sub_category in sub_categories_content.find_all('option'):
-            current_sub_category = self.verify_item_encoding((sub_category.get_text()).strip())
-            self.sub_categories[category_num].append(current_sub_category)
-            sub_categories.append(format(sub_category['value']))
-        return sub_categories
+            current_sub_category = sub_category.get_text().strip()
+            current_sub_category = self.verify_item_encoding(current_sub_category)
+            sub_categories.append({"Sub-Category-Name": current_sub_category, "Sub-Category-Link": format(sub_category['value'])})
+        self.sub_categories_dict.update({main_category: sub_categories})
```

Comment:
`sub_categories_dict` seems redundant.
You can change `main_categories_dict` to include this array of dicts